### PR TITLE
feat(printer): add support for YAML output

### PR DESCRIPTION
**Pull request by remyleone**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request.
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Relates OR Closes #0000

Release note for [CHANGELOG](https://github.com/scaleway/scaleway-cli/blob/v2/CHANGELOG.md):

```release-note

```

### CODE REVIEW

File: internal/core/printer.go [hunk 1]
```diff
@@ -199,3 +206,28 @@ func (p *Printer) printJSON(data interface{}) error {
...
+	}
+
+	writer := p.stdout
+	if isError {
+		writer = p.stderr
+	}
+	encoder := yaml.NewEncoder(writer)
+
+	// We handle special case to make sure that a nil slice is marshal as `[]`
+	if reflect.TypeOf(data).Kind() == reflect.Slice && reflect.ValueOf(data).IsNil() {
```
Comment (jerome-quere) [inline internal/core/printer.go:227]: Is it needed with yaml encoding ?

**Comment by remyleone** [comment]
About adding a test, would that mean adding YAML to all the goldens? just like JSON and human?

File: internal/core/printer.go (hunk 1 above)
Comment (remyleone) [reply internal/core/printer.go]: It doesn't seem to be required. I've removed it.

**Review by jerome-quere — APPROVED** [review]
LGTM

**Pull request merged**